pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -179,6 +179,11 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(builder *ListenerBui
 		}
 		listeners = append(listeners, ml.mutable.Listener)
 	}
+	if proxyConfig.GatewayTopology != nil && proxyConfig.GatewayTopology.ProxyProtocol != nil {
+		for _, listener := range listeners {
+			listener.ListenerFilters = append(listener.ListenerFilters, xdsfilters.ProxyProtocol)
```

Comment:
before we had INSERT_FIRST and now we are appending to the end? Do we want it first or last?

Reply:
also this is mostly a nitpick, but this may be a bit more clear in `buildListener` where we compute the other listenerFilters. Its a bit of a mess currently for sure, though.

Concretely, I think this would add the PROXY to QUIC as well - not sure if that is something we want or not? Can you do PROXY over HTTP/3? If we do want it, this is mostly a style nit so NBD

Reply:
Will change to INSERT_FIRST since that makes most sense.

PROXY can run on TCP over QUIC, but it won't work in Envoy because it's not inserted at the right layer. Envoy doesn't expose a TCP layer for QUIC in general.

Reply:
Made it insert_first, and removed from quic.